### ISSUE TITLE
TypeResolver depends on sun.misc.Unsafe which no longer exists in JDK11+

### ISSUE THREAD

**Issue by buko** [jhalterman/typetools#56]
This is probably why #52 , the test failing with OpenJDK 12 is happening. 

The error comes from [TypeResolver.java line 68](https://github.com/jhalterman/typetools/blob/master/src/main/java/net/jodah/typetools/TypeResolver.java#L68).

When that static intializer craps out the entire JVM dies. (Why static initializers are somewhat dangerous I think.)

```
Exception in thread "main" java.lang.NoClassDefFoundError: sun/misc/Unsafe
	at typetools@0.6.1/net.jodah.typetools.TypeResolver$1.run(TypeResolver.java:68)
	at typetools@0.6.1/net.jodah.typetools.TypeResolver$1.run(TypeResolver.java:65)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:551)
	at typetools@0.6.1/net.jodah.typetools.TypeResolver.<clinit>(TypeResolver.java:65)
```

**Comment by buko** [comment]
Short term work around, is bringing sun.misc.Unsafe back by adding the following to module.info:

```
requires typetools;
	requires jdk.unsupported; // HACK to make typetools work: See https://dzone.com/articles/using-sunmiscunsafe-in-java-9-1, https://github.com/jhalterman/typetools/issues/56
```

**Comment by jhalterman** [comment]
Would be happy to take a PR for this.

**Comment by mkurz** [comment]
@jhalterman Also fixed by #66?
